test/tests/components/ws-manager/prebuild_test.go

```diff
@@ -491,6 +408,263 @@ func TestOpenWorkspaceFromPrebuild(t *testing.T) {
 	testEnv.Test(t, f)
 }
 
+// TestPrebuildAndRegularWorkspaceDifferentWorkspaceClass
+// - create a prebuild with small workspace class (20Gi disk)
+// - create the workspace from prebulid with large workspace class (30Gi disk)
+// - make sure either one of the condition mets
+//   - the prebuild log message exists
+//   - the init task message exists
+//   - the init task generated file exists
+//
+// - make sure the .git/ folder with correct permission
+// - create a prebuild with large workspace class (30Gi disk) separately
+// - create the workspace from prebuild with small workspace class (20Gi disk) separately
+// - make sure the workspace can't start
+func TestPrebuildAndRegularWorkspaceDifferentWorkspaceClass(t *testing.T) {
+	f := features.New("prebuild").
+		WithLabel("component", "ws-manager").
+		Assess("it should open workspace with different workspace class", func(_ context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			tests := []struct {
+				Name                             string
+				PrebuildWorkspaceClass           string
+				RegularWorkspaceClass            string
+				ContextURL                       string
+				WorkspaceRoot                    string
+				CheckoutLocation                 string
+				ShouldFailToOpenRegularWorkspace bool
+			}{
+				{
+					Name: "prebuild-small-regular-large-workspace-class",
+					// TODO: do not use hard-code workspace class name to prevent if we change to different environment to run the test
+					PrebuildWorkspaceClass: "small",
+					RegularWorkspaceClass:  "default",
+					//
+					ContextURL:       "https://github.com/gitpod-io/empty",
+					CheckoutLocation: "empty",
+					WorkspaceRoot:    "/workspace/empty",
+				},
+				{
+					Name: "prebuild-large-regular-small-workspace-class",
+					// TODO: do not use hard-code workspace class name to prevent if we change to different environment to run the test
+					PrebuildWorkspaceClass: "default",
+					RegularWorkspaceClass:  "small",
+					//
```

Comment:
```suggestion
```

Reply:
I intended to add this line to know which parts are the TODO.